src/pip/_internal/req/req_uninstall.py

```diff
@@ -543,7 +543,7 @@ def from_dist(cls, dist):
 
         elif develop_egg_link:
             # develop egg
-            with open(develop_egg_link, 'r') as fh:
+            with open(develop_egg_link) as fh:
```

Comment:
What is encoding of egg_link file. What happen if non-ASCII path on Windows?

Reply:
It’s undefined because distutils is so ancient people didn’t care about this back then. Platform encoding is the backward compatible choice here.